### ISSUE TITLE
Ubuntu's version of rustc (1.25.0) can't build dependencies for ripgrep 0.10.0

### ISSUE THREAD

**Issue by x4121** [x4121/ripgrep-ubuntu#2]
Sadly Ubuntu's package repos (up to `bionic`) contain rustc in version 1.25.0, which seems to be too old to build ripgrep dependencies.

```
error[E0658]: use of unstable library feature 'thread_local_state': state querying was recently added (see issue #27716)
   --> vendor/parking_lot_core/src/parking_lot.rs:169:34
    |
169 |     if let Ok(tls) = THREAD_DATA.try_with(|x| x as *const ThreadData) {
    |                                  ^^^^^^^^
```

**Comment by BurntSushi** [comment]
Yup, unfortunately, you'll need to stick with ripgrep 0.9 until Ubuntu updates its Rust compiler.

**Comment by x4121** [comment]
Thanks for your confirmation (I'm not too deep into Rust and thought I may have gotten something wrong). Guess that change won't happen before ripgrep made it into Ubuntu, at which time this repo is (gladly) not useful anymore :slightly_smiling_face:

**Comment by BurntSushi** [comment]
@x4121 I actually think this repo could be incredibly useful! I'm certainly going to point folks to it who want an example of how to setup a ppa for distributing Rust applications. :)

**Comment by x4121** [comment]
Haha... damn, I was hoping some more skilled and experienced folks at Debian/Ubuntu would take responsibility for that 😂

I really do hope that I find a cleaner way of compiling Rust applications in the future. Especially when people could potentially use it as a reference :wink:
